src/zarr/codecs/pipeline.py

```diff
@@ -331,6 +331,7 @@ async def write_batch(
         value: NDBuffer,
         drop_axes: tuple[int, ...] = (),
     ) -> None:
+        write_empty_chunks = config.get("array.write_empty_chunks") == True  # noqa: E712
```

Comment:
> I'd rather make this a property of the Array so that we can guarantee consistent write behavior after an Array has been initialized.

If we do that, then we also require that users create a brand new array if they want to write just some parts of the data with different empty chunks handling (or we introduce `write_empty_chunks` as a mutable attribute, which I would rather avoid)

Reply:
i think this is resolved by making `write_empty_chunks` an attribute of the array config